pages/coinbase/homepage.py

```diff
@@ -1,45 +1,237 @@
-        self.header_get_started_button = Button(
-            self.driver, self.header_get_started_button_loc)
+    # A dismiss cookie dialogue box appears and hides elements, clicking
+    # the dismiss button performed in the init method fixes this issue  
+    visible_dialogue_box_loc =      (by.XPATH, "//div[@class='sc-fzoLsD gkNzOD']")
+    cookie_dismiss_button_loc =     (by.XPATH, "//button[.='Dismiss']")
 
+    # NAVIGATION BAR
 
+    home_link_loc =                 (by.XPATH, "//a[@title='Home']")
+    prices_link_loc =               (by.XPATH, "//a[@title='Prices']")
+    products_menu_loc =             (by.XPATH, "//button[contains(text(),'Products')]")
+    company_menu_loc =              (by.XPATH, "//button[contains(text(),'Company')]")
+    earn_crypto_link_loc =          (by.XPATH, "//a[@title='Earn crypto']")
+    sign_in_link_loc =              (by.XPATH, "//a[@title='Sign in']")
+    header_get_started_button_loc = (by.XPATH, "//a[.='Get started']")
+
+         
+    # CREATE ACCOUNT
+
+    get_started_email_field_loc =   (by.XPATH, "//input[@type='email']")
+    get_started_button_loc =        (by.XPATH, "//input[@type='email']/following-sibling::button")
+
+    # CRYPTO CURRENCY MENU
+
+    bitcoin_text_link_loc =         (by.XPATH, "//h4[.='Bitcoin']")
+    bitcoin_BTC_link_loc =          (by.XPATH, "//h4[.='BTC']")
+    bitcoin_logo_link_loc =         (by.XPATH, "//img[@alt='Bitcoin logo']")
+    bitcoin_price_button_loc =      (by.XPATH, "//a[@href='/price/bitcoin']/button")
+
+    ethereum_text_link_loc =        (by.XPATH, "//h4[.='Ethereum']")
+    ethereum_ETH_link_loc =         (by.XPATH, "//h4[.='ETH']") 
+    ethereum_logo_link_loc =        (by.XPATH, "//img[@alt='Ethereum logo']")
+    ethereum_price_button_loc =     (by.XPATH, "//a[@href='/price/ethereum']/button")
+
+    bitcoinCash_text_link_loc =     (by.XPATH, "//h4[.='Bitcoin Cash']")        
+    bitcoinCash_BCH_link_loc =      (by.XPATH, "//h4[.='BCH']")
+    bitcoinCash_logo_link_loc =     (by.XPATH, "//img[@alt='Bitcoin Cash logo']")
+    bitcoinCash_price_button_loc =  (by.XPATH, "//a[@href='/price/bitcoin-cash']/button")
+
+    litecoin_text_link_loc =        (by.XPATH, "//h4[.='Litecoin']")
+    litecoin_LTC_link_loc =         (by.XPATH, "//h4[.='LTC']")
+    litecoin_logo_link_loc =        (by.XPATH, "//img[@alt='Litecoin logo']")
+    litecoin_price_button_loc =     (by.XPATH, "//a[@href='/price/litecoin']/button")
+
+    # EARNING LINKS BLCOK
+
+    start_earning_button_loc =       (by.XPATH, "//button[contains(@class,'EarnUpsellBanner')]")
+    start_earning_lower_button_loc = (by.XPATH, "//button[contains(@class,'EarnFooterBanner')]")
+    maker_link_loc =                 (by.XPATH, "//a[@href='/earn/maker']/div")
+    celo_link_loc =                  (by.XPATH, "//a[@href='/earn/celo']/div/div/h2")   #I don't love this xpath, but I feel it is the best way to locate this element
```

Comment:
Why not just `//a[@to='/earn/celo']`? I mean that's also bad xpath for the reasons in my rant above, but given the circumstances I'll allow it. Why do you dig all the way down to the h2 element when you could just click the anchor?

Reply:
I think I forgot to go back and fix this xpath after I fixed the cookie banner getting in the way. So the anchor is what you call the highest parent level of an element?